File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    let hash = {}
    name.forEach((v,i)=> hash[v] = yearning[i]);
    
    return photo.map(v=>v.reduce((a,c)=> !hash[c] ? a+0 : a+hash[c],0))
}   

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        
        HashMap<String,Integer> map = new HashMap<>();

        for (int i = 0; i < name.length; i++) {
            map.put(name[i], yearning[i]);
        }

        int[] answer = Arrays.stream(photo)
                .mapToInt(p -> Arrays.stream(p).mapToInt(n-> map.getOrDefault(n,0)).sum())
                .toArray();
        
        return answer;
    }
}